mev_inspect/classifiers/helpers.py

```diff
@@ -84,3 +96,69 @@ def _filter_transfers(
         filtered_transfers.append(transfer)
 
     return filtered_transfers
+
+
+def is_valid_0x_swap(
+    trace: DecodedCallTrace,
+    child_transfers: List[Transfer],
+) -> bool:
+
+    # 1. There should be 2 child transfers, one for each settled leg of the order
+    if len(child_transfers) != 2:
+        raise ValueError(
+            f"A settled order should consist of 2 child transfers, not {len(child_transfers)}."
+        )
+
+    # 2. The function signature must be in the lists of supported signatures
+    if trace.function_signature not in (LIMIT_SIGNATURES + RFQ_SIGNATURES):
+        raise RuntimeError(
+            f"0x orderbook function {trace.function_signature} is not supported"
+        )
+
+    # 3. The swap should not be a child of previous swaps in the block
+
+    return True
+
+
+def _get_taker_token_in_amount(
+    taker_address: str, token_in_address: str, child_transfers: List[Transfer]
+) -> int:
+
+    if taker_address == ANY_TAKER_ADDRESS:
+        for transfer in child_transfers:
+            if transfer.token_address == token_in_address:
+                return transfer.amount
+    else:
+        for transfer in child_transfers:
+            if transfer.to_address == taker_address:
+                return transfer.amount
+    return 0
+
+
+def get_0x_token_in_data(
+    trace: DecodedCallTrace, child_transfers: List[Transfer]
+) -> Tuple[str, int]:
+
+    order: List = trace.inputs["order"]
+    token_in_address = order[0]
+
+    if trace.function_signature in RFQ_SIGNATURES:
```

Comment:
right now you're checking if `trace.function_signature` is in `RFQ_SIGNATURES` or `LIMIT_SIGNATURES` in another function and then _assuming_ here that whoever calls this function also remembered to call that one

Better to throw in the "else" clause here (where you're using that assumption) than have the check elsewhere

Reply:
Gotcha